src/golang/cmd/server/handler/register_airflow_workflow.go

```diff
@@ -120,6 +137,45 @@ func (h *RegisterAirflowWorkflowHandler) Perform(ctx context.Context, interfaceA
 
 	emptyResp := registerAirflowWorkflowResponse{}
 
+	if args.isUpdate {
+		// Sync existing Airflow DAGRuns before DAG is updated
+		workflowDag, err := utils.ReadLatestWorkflowDagFromDatabase(
+			ctx,
+			args.dbWorkflowDag.WorkflowId,
+			h.WorkflowReader,
+			h.WorkflowDagReader,
+			h.OperatorReader,
+			h.ArtifactReader,
+			h.WorkflowDagEdgeReader,
+			h.Database,
+		)
+		if err != nil {
+			return emptyResp, http.StatusInternalServerError, errors.Wrap(err, "Unable to update workflow.")
+		}
+
+		// NOTE (saurav): This is not perfect because if there are any in progress Airflow DAGRuns, those will
```

Comment:
TODO: Create linear task for this

Reply:
I'm curious if this is caused by us restricting sync to finished runs here: https://github.com/aqueducthq/aqueduct/blob/main/src/golang/lib/airflow/writer.go#L28 , or something else?

Does removing that restriction solve the problem without any additional concerns?

Reply:
It will actually add more complexity, because we then need to check all workflow dag results to see if there are any with a status of in progress and then try to update their status on each subsequent sync call.

Reply:
Ahh understood. Then agreed what we have now is good enough